CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/markers/PolygonMarker.java

```diff
@@ -0,0 +1,144 @@
+    private String rawPath = "";
+
+    public PolygonMarker(final MapView parent, final int markerID, final int xOffset, final int yOffset) {
+        super(parent, markerID, -99, xOffset, yOffset, AbstractMarker.MarkerType.POLYGON_MARKER);
+
+        markerPath.setStroke(Color.BLACK);
+        markerPath.setFill(Color.ORANGE);
+        markerPath.setOpacity(0.4);
+
+        // Event handler for the polygon marker
+        markerPath.setOnMouseEntered((final MouseEvent e) -> {
+            markerPath.setFill(Color.YELLOW);
+
+            e.consume();
+        });
+
+        markerPath.setOnMouseExited((final MouseEvent e) -> {
+            markerPath.setFill(Color.ORANGE);
+
+            e.consume();
+        });
+
+        markerPath.setOnMouseClicked((final MouseEvent e) -> {
+            parent.removeUserMarker(markerID);
+            e.consume();
+        });
+    }
+
+    /**
+     * Adds a new line to the polygon marker
+     *
+     * @param prevLineEndX
+     * @param prevLineEndY
+     * @return
+     */
+    public Line addNewLine(final double prevLineEndX, final double prevLineEndY) {
+        // If current line is null it mean the first line is being drawn
+        // Else finish off the current line and add it to the UI
+        if (currentLine == null) {
+            currentLine = new Line();
+            currentLine.setStartX(prevLineEndX);
+            currentLine.setStartY(prevLineEndY);
+            setEnd(prevLineEndX, prevLineEndY);
+        } else {
+            setEnd(prevLineEndX, prevLineEndY);
+            polygonLineUI.add(currentLine);
+            currentLine = new Line();
+            currentLine.setStartX(prevLineEndX);
+            currentLine.setStartY(prevLineEndY);
+            setEnd(prevLineEndX, prevLineEndY);
+        }
+        currentLine.setStroke(Color.BLACK);
+        return currentLine;
+    }
+
+    /**
+     * Clear the gui that shows the polygon the user is drawing
+     */
+    public void endDrawing() {
+        currentLine = null;
+        polygonLineUI.clear();
+    }
+
+    public void setEnd(final double x, final double y) {
+        if (currentLine != null) {
+            currentLine.setEndX(x);
+            currentLine.setEndY(y);
+        }
+    }
+
+    /**
+     * Generate the polygon SVG shape
+     */
+    public void generatePath() {
+        String path = "";
+
+        // generate raw path based on vertices
+        if (!polygonLineUI.isEmpty()) {
+            for (int i = 0; i < polygonLineUI.size(); i++) {
+                if (i == 0) {
+                    path = "M" + polygonLineUI.get(i).getStartX() + SeparatorConstants.COMMA + polygonLineUI.get(i).getStartY();
+                    path += "L" + polygonLineUI.get(i).getEndX() + SeparatorConstants.COMMA + polygonLineUI.get(i).getEndY();
+                } else {
+                    path += "L" + polygonLineUI.get(i).getEndX() + SeparatorConstants.COMMA + polygonLineUI.get(i).getEndY();
+                }
```

Comment:
take this line out of the if and else block (since it is at the end of both). Could therefore remove the else block altogether